rules/TypeDeclaration/Rector/ClassMethod/StrictStringParamConcatRector.php

```diff
@@ -153,6 +159,14 @@ private function resolveVariableConcattedFromParam(
         return $variableConcatted;
     }
 
+    private function isVariableInArg(Arg $arg, string $paramName): bool
+    {
+        return (bool) $this->betterNodeFinder->findFirst(
+            $arg->value,
+            fn (Node $subNode): bool => $subNode instanceof Variable && $this->isName($subNode, $paramName)
+        );
+    }
```

Comment:
@TomasVotruba I think we can move this handling to a guard service to be reused when parameter used as Arg value in diferent rules, another rule that has same issue is:

- https://github.com/rectorphp/rector/issues/8091

Reply:
Variable Param reassign check probably better instead of Arg check, I will check tomorrow.

Reply:
https://github.com/rectorphp/rector-src/pull/4775